### DISCUSSION TITLE
Tracking User Events with the Storefront API

### DISCUSSION THREAD

**Discussion by jackmcosta32** [Shopify/storefront-api-feedback#111] [Help]
Hey all, I'm currently a developer at Wine Vinhos S.A, and I'd like to ask some questions about the Shopify analytics API.

We are considering migrating our current liquid store to a ReactJS-based approach that would still be able to track the user events through Shopify. Because of that, we have taken two approaches into account, one of them is to redevelop our application using the Hydrogen framework (https://shopify.dev/custom-storefronts/hydrogen/framework/analytics). The other one is to emit events through the admin API (https://shopify.dev/api/admin-rest/2022-04/resources/event#top). About those approaches, we'd like to know about the following:

- Do you guys know if the admin events API is able to emit user events like PAGE_VIEW, ADD_TO_CART, and others that are documented on the Hydrogen framework docs page?

- About Hydrogen, we'd like to know if there is any sort of documentation containing all of the events supported by the useServerAnalytics hook.

Thanks :wink:

**Comment by thomasKn** [comment]
Hey @jackmcosta32 did you find a way to emit `PAGE_VIEW` and  `ADD_TO_CART` events?

**Reply by jackmcosta32** [reply]
Hi there @thomas, sorry for taking so long to answer. As Hydrogen was in its early days when this issue was opened, we thought it would be better to keep a standalone next-js application with a CDP like segment or rudderstack for handling events, and communicating with Shopify API through a BFF.

**Reply by thomasKn** [reply]
Thanks @jackmcosta32 makes sense 👍

**Comment by davidhousedev** [comment]
Im not a Shopify Employee, just following along!

These features have been built into Hydrogen, specifically into `hydrogen-react`.

Bear in mind that work to integrate with analytics today may need to be migrated if you want to move from Hydrogen v1 to v2 which is expected to be released soon.

Check out these docs: https://shopify.dev/custom-storefronts/hydrogen/analytics
